drivers/storage/portworx/component/px_resource_gateway.go

```diff
@@ -0,0 +1,444 @@
+		},
+	}
+	return k8sutil.CreateOrUpdateClusterRole(
+		p.k8sClient,
+		&rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: PxResourceGatewayClusterRoleName,
+			},
+			Rules: rules,
+		},
+	)
+}
+
+func (p *pxResourceGateway) createClusterRoleBinding(
+	clusterNamespace string,
+) error {
+	return k8sutil.CreateOrUpdateClusterRoleBinding(
+		p.k8sClient,
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: PxResourceGatewayClusterRoleBindingName,
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      PxResourceGatewayServiceAccountName,
+					Namespace: clusterNamespace,
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				Kind:     "ClusterRole",
+				Name:     PxResourceGatewayClusterRoleName,
+				APIGroup: "rbac.authorization.k8s.io",
+			},
+		},
+	)
+}
+
+func (p *pxResourceGateway) createDeployment(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	imageName := p.getDesiredPxResourceGatewayImage(cluster)
+	if imageName == "" {
+		return fmt.Errorf("pxResourceGateway image cannot be empty")
+	}
+
+	existingDeployment := &appsv1.Deployment{}
+	err := p.k8sClient.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      PxResourceGatewayDeploymentName,
+			Namespace: cluster.Namespace,
+		},
+		existingDeployment,
+	)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	targetCPU := defaultPxResourceGatewayCPU
+	if cpuStr, ok := cluster.Annotations[pxutil.AnnotationPxResourceGatewayCPU]; ok {
+		targetCPU = cpuStr
+	}
+	targetCPUQuantity, err := resource.ParseQuantity(targetCPU)
+	if err != nil {
+		return err
+	}
+
+	targetCPULimitQuantity, err := resource.ParseQuantity(defaultPxResourceGatewayCPULimit)
+	if err != nil {
+		return err
+	}
+
+	args := map[string]string{}
+	for k, v := range cluster.Spec.PxResourceGateway.Args {
+		if _, exists := autopilotConfigParams[k]; exists {
+			continue
+		}
+		key := strings.TrimLeft(k, "-")
+		if len(key) > 0 && len(v) > 0 {
+			args[key] = v
+		}
+	}
+
+	argList := make([]string, 0)
+	for k, v := range args {
+		argList = append(argList, fmt.Sprintf("--%s=%s", k, v))
+	}
+	sort.Strings(argList)
+	command := append([]string{"/px-resource-gateway"}, argList...)
+
+	imageName = util.GetImageURN(cluster, imageName)
+
+	var existingImage string
+	var existingCommand []string
+	var existingEnvs []v1.EnvVar
+	for _, c := range existingDeployment.Spec.Template.Spec.Containers {
+		if c.Name == PxResourceGatewayContainerName {
+			existingImage = c.Image
+			existingCommand = c.Command
+			existingEnvs = append([]v1.EnvVar{}, c.Env...)
+			sort.Sort(k8sutil.EnvByName(existingEnvs))
+			break
+		}
+	}
+
+	targetDeployment := p.getPxResourceGatewayDeploymentSpec(cluster, ownerRef, imageName,
+		command, targetCPUQuantity, targetCPULimitQuantity)
+
+	// Check if the deployment has changed
+	modified := existingImage != imageName ||
+		!reflect.DeepEqual(existingCommand, command) ||
```

Comment:
Instead of comparing each and every part of the deployment spec, we made some recent changes to compare the pod template directly. 
Check this out for reference - https://github.com/libopenstorage/operator/blob/release-24.1.2-drw/drivers/storage/portworx/component/csi.go#L462